app/cdap/components/PipelineDiff/DiffCanvas/PluginNode.tsx

```diff
@@ -0,0 +1,156 @@
+import { IPipelineNodeData } from '../store/diffSlice';
+import { objectQuery } from 'services/helpers';
+import { pluginDiffColors } from '../util/helpers';
+import { DiffIcon } from '../DiffIcon';
+
+const NodeRoot = styled.div`
+  height: 110px;
+  width: 200px;
+  border: 3px solid ${(props) => pluginDiffColors[props.diffIndicator]};
+  border-radius: 5px;
+  background: white;
+  display: flex;
+  flex-direction: column;
+`;
+
+const HeaderRoot = styled.div`
+  width: 100%;
+  display: flex;
+`;
+
+const CustomIconRoot = styled.img`
+  &&& {
+    width: 25px;
+    height: 25px;
+    margin: 10px;
+  }
+`;
+
+const IconRoot = styled.div`
+  &&& {
+    font-size: 25px;
+    padding: 10px;
+  }
+`;
+
+const TitleContainer = styled.div`
+  flex-grow: 1;
+  padding-top: 10px;
+`;
+
+const TitleLabel = styled.label`
+  display: block;
+  color: ${(props) => pluginDiffColors[props.diffIndicator]};
+  font-size: 14px;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const SubtitleLabel = styled(TitleLabel)`
+  font-size: 10px;
+  color: ${(props) => pluginDiffColors[props.diffIndicator] + '88'};
+`;
+
+const DiffIconContainer = styled.div`
+  display: flex;
+  flex-grow: 1;
+  justify-content: flex-end;
+  align-items: flex-end;
+  padding: 0 10px 10px 10px;
+`;
+
+const HandleRoot = styled(Handle)`
+  width: 14px;
+  height: 14px;
+  background-color: #4e5568;
+  border-radius: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const AlertHandleRoot = styled(HandleRoot)`
+  left: 30px;
+`;
+
+const ErrorHandleRoot = styled(HandleRoot)`
+  left: 60px;
+`;
+
+const BottomHandleLabel = styled.span`
+  font-size: 11px;
+  position: absolute;
+  top: -16px;
+  color: ${(props) => pluginDiffColors[props.diffIndicator] + '40'};
+`;
+
+const PluginNode = ({
+  id,
+  data,
+  children,
+}: React.PropsWithChildren<NodeProps<IPipelineNodeData>>) => {
+  const diffIndicator = objectQuery(data, 'diffItem', 'diffIndicator') ?? 'default';
+  return (
+    <NodeRoot id={id} diffIndicator={diffIndicator}>
+      <HeaderRoot>
+        {data.customIconSrc ? (
+          <CustomIconRoot src={data.customIconSrc} />
+        ) : (
+          <IconRoot className={classnames('node-icon fa', data.iconName)} />
+        )}
+        <TitleContainer>
+          <TitleLabel diffIndicator={diffIndicator}>{data.plugin.label}</TitleLabel>
+          <SubtitleLabel diffIndicator={diffIndicator}>
+            {data.plugin.artifact.version}
+          </SubtitleLabel>
+        </TitleContainer>
+      </HeaderRoot>
+      <DiffIconContainer>
+        {diffIndicator !== 'default' && <DiffIcon diffType={diffIndicator} fontSize="large" />}
+      </DiffIconContainer>
+      <HandleRoot type="target" position={Position.Left} />
+      <HandleRoot type="source" id="source_right" position={Position.Right} />
+      {children}
+    </NodeRoot>
+  );
+};
+
+export const DefaultPluginNode = (props: NodeProps<IPipelineNodeData>) => {
```

Comment:
What is the point of this component? all it does is return pluginNode with the same props?

Reply:
updated